src/components/widget/DataLinks.jsx

```diff
@@ -3,6 +3,11 @@ import { Text } from '../elements/Text';
 import { Button } from '../elements/Button';
 
 function DataLinks({ shortUrl, originalUrl }) {
+  function copyToClipboard(text) {
+    navigator.clipboard.writeText(text).catch((err) => {
+      console.error('Failed to copy data', err);
+    });
+  }
```

Comment:
Man, you should start creating something static inside of components.

Look, the function copyToClipboard will be created from scratch each time a component rerender happens. It might be okay, when your function depends on some variables from the component scope, but in your case it takes all what's needed as a param.

So, just move it away from the scope of component. You can easily store it in helpers.